### DISCUSSION TITLE
Loading local stl files

### DISCUSSION THREAD

**Discussion by jliu309** [zauberzeug/nicegui#1482] [Q&A]
### Description

I am trying to load a local .stl file of a UR robot to `ui.scene` such that it can be visualized. Following the example in [this page](https://nicegui.io/documentation/scene), I tried the following:

```py
from nicegui import ui

with ui.scene().classes('w-full h-64') as scene:
    teapot = 'file:///home/user1/.../ur5.stl'
    scene.stl(teapot)

ui.run()
```
But nothing showed up in the scene. I have struggled for this for hours but still don't know how to resolve this. Any help will be greatly appreciated!

**Comment by natankeddem** [comment]
To make local files available for hosting on the webserver you need to map them.
https://nicegui.io/documentation#static_files
```py
from nicegui import app, ui

app.add_static_files('/stlfiles', 'stlfiles')

with ui.scene().classes('w-full h-64') as scene:
    teapot = '/stlfiles/Utah_teapot_(solid).stl'
    scene.stl(teapot)
ui.run()
```
